tools/ci-scripts/windows/Install-Boost.ps1

```diff
@@ -0,0 +1,172 @@
+
+#>
+[CmdletBinding(SupportsShouldProcess=$True)]
+param
+(
+    [string]
+    $Version,
+
+    [ValidateSet('12.0', '14.0', '14.1')]
+    [string]
+    $VsNum,
+    
+    [string[]]
+    $Components = (
+        'boost_chrono',
+        'boost_date_time',
+        'boost_thread',
+        'boost_system',
+        'boost_unit_test_framework'),
+
+    [string]
+    $OutputDirectory = $null
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+function Get-TempFolder
+{
+    $base = [System.IO.Path]::GetTempPath()
+    $result = ''
+
+    do
+    {
+        $randomPart = ''
+        while ($randomPart.Length -lt 5)
+        {
+            $asciiBase = 97 # ASCII for 'a'
+
+            $r = Get-Random -Minimum 0 -Maximum 36
+            if ($r -ge 26) # We use [26-35] for numbers
+            {
+                $asciiBase = 48 # ASCII for '0'
+                $r -= 26 # adjust to be in [0-9] from [26-36].
+            }
+
+            $randomPart += [char]($r + $asciiBase)
+        }
+
+        $result = [System.IO.Path]::Combine($base, $randomPart)
+    } while (Test-Path -LiteralPath $result -PathType Container)
+
+    # There's a TOCTTOU race in this check, but it's very unlikely.
+    Write-Debug "Generated temp folder name: '$result'"
+    return $result
+}
+
+function ConvertVsNum-ToBoostPackageFormat([string]$vsNum)
+{
+    return $vsNum.Replace('.', '')
+}
+
+function Install-NuGetPackage([string]$InstallDir, [string]$PackageId, [string]$PackageVersion)
+{
+    $nugetPrereleaseFlag = @()
+    if ($packageVersion.Contains('-')) {
+        $nugetPrereleaseFlag += 'Prerelease'
+    }
+
+    nuget install $packageId `
+        -OutputDirectory $installDir `
+        -Version $packageVersion `
+        -ExcludeVersion `
+        $nugetPrereleaseFlag
+
+    if (-not $?)
+    {
+        throw "NuGet install of '$packageId' '$packageVersion' failed. Exit code $LastExitCode"
+    }
+}
+
+if (-not $OutputDirectory)
+{
+    $OutputDirectory = Get-TempFolder
+}
+
+if (-not (Test-Path -LiteralPath $OutputDirectory -PathType Container))
+{
+    mkdir $OutputDirectory | Out-Null
+}
+
+$workDir = [System.IO.Path]::Combine($OutputDirectory, 'w')
+$destDir = [System.IO.Path]::Combine($OutputDirectory, 'd')
+$lib32Dir = [System.IO.Path]::Combine($destDir, 'lib32')
+$lib64Dir = [System.IO.Path]::Combine($destDir, 'lib64')
+mkdir $workDir | Out-Null
+mkdir $destDir | Out-Null
+mkdir $lib32Dir | Out-Null
+mkdir $lib64Dir | Out-Null
+
+function Install-BoostHeaders
+{
+    Write-Progress -Activity 'Installing Boost' -Status "Installing 'boost' (headers)"
+
+    if ($PSCmdlet.ShouldProcess('boost', 'Install NuGet package'))
+    {
+        Install-NuGetPackage `
+            -PackageId 'boost' `
+            -InstallDir $workDir `
+            -PackageVersion $Version
+
+        Move-Item `
+            -Path ([System.IO.Path]::Combine($workDir, 'boost', 'lib', 'native', 'include', 'boost')) `
```

Comment:
Is there any benefit in using the sub-folder names separately instead of a simple `boost\lib\native\include\boost`?